pkg/kafka/streaming.go

```diff
@@ -0,0 +1,117 @@
+package kafka
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/kafka/v1beta1"
+)
+
+// Persistent returns the custom resource for a persistent Kafka
+// Reference: https://github.com/strimzi/strimzi-kafka-operator/blob/master/examples/kafka/kafka-persistent.yaml
+func Persistent(jaeger *v1.Jaeger) v1beta1.Kafka {
+	trueVar := true
+	return v1beta1.Kafka{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jaeger.Name,
+			Namespace: jaeger.Namespace,
+			Labels: map[string]string{
+				"app":                         "jaeger",
+				"app.kubernetes.io/name":      jaeger.Name,
+				"app.kubernetes.io/instance":  jaeger.Name,
+				"app.kubernetes.io/component": "kafkauser",
+				"app.kubernetes.io/part-of":   "jaeger",
+
+				// workaround for https://github.com/strimzi/strimzi-kafka-operator/issues/2107
+				"app.kubernetes.io/managed---by": "jaeger-operator",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				metav1.OwnerReference{
+					APIVersion: jaeger.APIVersion,
+					Kind:       jaeger.Kind,
+					Name:       jaeger.Name,
+					UID:        jaeger.UID,
+					Controller: &trueVar,
+				},
+			},
+		},
+		Spec: v1beta1.KafkaSpec{
+			v1.NewFreeForm(map[string]interface{}{
+				"kafka": map[string]interface{}{
+					"version":  "2.3.0",
```

Comment:
Does the version need to be supplied? It would be better if the operator was put in charge of choosing the most appropriate version - in the same way that the jaeger-operator does for jaeger backend.